src/routes/tenant/index.ts

```diff
@@ -0,0 +1,114 @@
+import { FastifyInstance, RequestGenericInterface } from 'fastify'
+import { FromSchema } from 'json-schema-to-ts'
+import { pool } from '../../utils/multitenant-db'
+import { cacheTenantConfigAndRunMigrations, deleteTenantConfig } from '../../utils/tenant'
+
+const schema = {
+  body: {
+    type: 'object',
+    required: ['anonKey', 'databaseUrl', 'jwtSecret', 'serviceKey'],
+    properties: {
+      anonKey: { type: 'string' },
+      databaseUrl: { type: 'string' },
+      jwtSecret: { type: 'string' },
+      serviceKey: { type: 'string' },
+    },
+  },
+} as const
+
+interface tenantRequestInterface extends RequestGenericInterface {
+  Body: FromSchema<typeof schema.body>
+  Params: {
+    tenantId: string
+  }
+}
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export default async function routes(fastify: FastifyInstance) {
+  fastify.get('/', async () => {
```

Comment:
How are these admin routes protected? Can we have a simple key based auth here?

Reply:
Added auth based on `apikey` header.